napari/_vispy/vispy_base_layer.py

```diff
@@ -195,3 +206,26 @@ def on_draw(self, event):
         """Called whenever the canvas is drawn.
         """
         self.layer.scale_factor = self.scale_factor
+
+
+def get_max_texture_sizes():
+    """Get maximum texture sizes for 2D and 3D rendering.
+    Returns
+    -------
+    MAX_TEXTURE_SIZE_2D : int or None
+        Max texture size allowed by the vispy canvas during 2D rendering.
+    MAX_TEXTURE_SIZE_3D : int or None
+        Max texture size allowed by the vispy canvas during 2D rendering.
+    """
+    # A canvas must be created to access gl values
+    _ = Canvas(show=False)
+    MAX_TEXTURE_SIZE_2D = gl.glGetParameter(gl.GL_MAX_TEXTURE_SIZE)
+    if MAX_TEXTURE_SIZE_2D == ():
+        MAX_TEXTURE_SIZE_2D = None
+    # vispy doesn't seem to expose this so hard coding for now ....
+    # MAX_TEXTURE_SIZE_3D = gl.glGetParameter(gl.GL_MAX_3D_TEXTURE_SIZE)
+    MAX_TEXTURE_SIZE_3D = 2048
+    if MAX_TEXTURE_SIZE_3D == ():
+        MAX_TEXTURE_SIZE_3D = None
```

Comment:
This logic seems weird... Did you mean to replace None with 2048 here? Or...?

Reply:
No, the `MAX_TEXTURE_SIZE_3D = 2048` is meant to replace call to the `gl.glGetParameter` that is commented above until I can figure out how to make that work. The `== ()` is only needed in the case that we can get that working, but it seemed ok to have in there. Maybe a little weird though.

I changed the comments slightly so maybe it makes more sense now, but happy to do different if you recommend

Reply:
i have now improved commented region so the code that won't execute is commented